the-collab-lab/tcl-30-whats-near-me
# [TCL30-38] Share Location Button

## Description

As a user I want the map to show pins near my location So that I can discover interesting things around me



## Related Issue

Closes [TCL30-13](https://the-collab-lab.atlassian.net/browse/TCL30-13)

## Acceptance Criteria

- Provide a prompt within the app (e.g., a button) for the user to tap/click to start sharing their location
- The browser will mostly likely prompt the user to accept (or not) sharing of their device location with the app
- If the user accepts, we…
   * use the returned location to set the center of the map
   * show the user as a dot or other marker on the map
   * fetch new pins based on the new center
   * show the user a new prompt to stop sharing their location (we will handle the case where the user stops sharing their location in TCL20-22)

- If the user does not accept, we…
  * leave the map in whatever is its last good state
  * leave the prompt to share their location active
## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

<img width="1037" alt="TCL30-38_before" src="https://user-images.githubusercontent.com/38255956/127882610-4e2c3e70-25e9-4b83-ac01-c3678e3c53e9.png">

### After

https://user-images.githubusercontent.com/38255956/127887688-9256afa3-964b-41dd-ad9c-234240cc3ddb.mov







## Testing Steps / QA Criteria



1. Clone repository: `git clone https://github.com/the-collab-lab/tcl-30-whats-near-me`.
2. Change to PR branch: `git checkout jw-ac-mm-share-location-button`
3. Install NPM dependencies: `npm install`.
4. Run dev server: `npm run start`
5. Open your browser: `http://localhost:3000/`
6. When a user loads the / screen view (can also be considered the Home view). There should be a button That asks user if they would like to share their current location. 
7. Once they share our location, we need to be able to show pins for the shared location.
